### PR TITLE
HIVE-4525 Support timestamps earlier than 1970 and later than 2038

### PR DESCRIPTION
**Pull request by mbautin**
@jhartlaub

### CODE REVIEW

File: serde/src/java/org/apache/hadoop/hive/serde2/io/TimestampWritable.java [hunk 1]
```diff
@@ -310,7 +357,7 @@ public int compareTo(TimestampWritable t) {
       }
       return n1 - n2;
```
Comment (jhartlaub) [inline serde/src/java/org/apache/hadoop/hive/serde2/io/TimestampWritable.java:358]: minor nit: why not the same as below (-1, 1) rather than a magnitude?

Comment (jhartlaub) [inline serde/src/java/org/apache/hadoop/hive/serde2/io/TimestampWritable.java:358]: ah, "0" too, nevermind

Comment (mbautin) [inline serde/src/java/org/apache/hadoop/hive/serde2/io/TimestampWritable.java:358]: In this case `n1` and `n2` are guaranteed to be between 0 and 999999999, allowing for a simplified expression.

**Comment by jhartlaub** [comment]
Looks good.

**Pull request merged**